### ISSUE TITLE
Databricks - "NOT NULL" combined with "GENERATED" in column definitions is reported as unparsable by sqlfluff but is valid Databricks syntax

### ISSUE THREAD

**Issue by david-mcdowell-ilw** [sqlfluff/sqlfluff#7024]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Although the [Databricks syntax for generated columns](https://docs.databricks.com/aws/en/delta/generated-columns) is supposed to be supported by sqlfluff already, it does not appear to work when using the NOT NULL syntax either preceding or following that phrase.

Given a test.sql file of:

```sql
CREATE TABLE data_engineering.test_sqlfluff_generated
(
    -- sql fluff accepts
    test_generated_always_as_id    BIGINT GENERATED ALWAYS AS IDENTITY
    , test_generated_default_as_id BIGINT GENERATED BY DEFAULT AS IDENTITY
    -- sql fluff does not accept
    , test_generated_always_as_id_not_null  BIGINT GENERATED ALWAYS AS IDENTITY NOT NULL
    , test_generated_default_as_id_not_null BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL
    , test_not_null_generated_always_as_id  BIGINT NOT NULL GENERATED ALWAYS AS IDENTITY
    , test_not_null_generated_default_as_id BIGINT NOT NULL GENERATED BY DEFAULT AS IDENTITY
)
USING DELTA;
```
SQLFluff reports this syntax as unparsable:
```bash
$ sqlfluff lint --config ../../pyproject.toml test.sql
== [test.sql] FAIL                                                                                                                                                                                 
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: '(\n    --
                       | sql fluff accepts\n    test_gene...'
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

When I comment out the "unparsable syntax" lines, the file is linted successfully:
```sql
CREATE TABLE data_engineering.test_sqlfluff_generated
(
    -- sql fluff accepts
    test_generated_always_as_id    BIGINT GENERATED ALWAYS AS IDENTITY
    , test_generated_default_as_id BIGINT GENERATED BY DEFAULT AS IDENTITY
    -- sql fluff does not accept
    --, test_generated_always_as_id_not_null  BIGINT GENERATED ALWAYS AS IDENTITY NOT NULL
    --, test_generated_default_as_id_not_null BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL
    --, test_not_null_generated_always_as_id  BIGINT NOT NULL GENERATED ALWAYS AS IDENTITY
    --, test_not_null_generated_default_as_id BIGINT NOT NULL GENERATED BY DEFAULT AS IDENTITY
)
USING DELTA;
```
```bash
sqlfluff lint --config ../../pyproject.toml test.sql
All Finished 📜 🎉!
```

### Expected Behaviour

I expect sqlfluff to report this as valid, as this is valid Databricks SQL and a table can be created using the same .sql file:

<img width="808" height="727" alt="Image" src="https://github.com/user-attachments/assets/4aa26a45-f493-4ade-b782-b938ec297637" />

Databricks SQL version 2025.16:

<img width="255" height="288" alt="Image" src="https://github.com/user-attachments/assets/0f19b43c-a44c-45ba-b76a-17cb159e7658" />

### Observed Behaviour

SQLFluff reports this syntax as unparsable:
```bash
$ sqlfluff lint --config ../../pyproject.toml test.sql
== [test.sql] FAIL                                                                                                                                                                                 
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: '(\n    --
                       | sql fluff accepts\n    test_gene...'
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Given a test.sql file of:

```sql
CREATE TABLE data_engineering.test_sqlfluff_generated
(
    -- sql fluff accepts
    test_generated_always_as_id    BIGINT GENERATED ALWAYS AS IDENTITY
    , test_generated_default_as_id BIGINT GENERATED BY DEFAULT AS IDENTITY
    -- sql fluff does not accept
    , test_generated_always_as_id_not_null  BIGINT GENERATED ALWAYS AS IDENTITY NOT NULL
    , test_generated_default_as_id_not_null BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL
    , test_not_null_generated_always_as_id  BIGINT NOT NULL GENERATED ALWAYS AS IDENTITY
    , test_not_null_generated_default_as_id BIGINT NOT NULL GENERATED BY DEFAULT AS IDENTITY
)
USING DELTA;
```

and a pyproject.toml containing:
```toml
[tool.sqlfluff.core]
# https://docs.sqlfluff.com/en/stable/reference/dialects.html
# ANSI is best for a general setting that is compatible with most SQL dialects.
# Modify this value if using a specific SQL dialect, e.g., databricks, postgres, sparksql, etc.
dialect = "databricks"  # For this project, we are using Databricks
large_file_skip_byte_limit = 0  # do not skip large files
max_line_length = 120
sql_file_exts = ".sql,.sql.j2,.dml,.ddl,.hql"

[tool.sqlfluff.indentation]
allow_implicit_indents = true
indented_on_contents = false

[tool.sqlfluff.layout.type.alias_expression]
spacing_before = "align"  # non-default spacing before alias expression
align_within = "select_clause"
align_scope = "bracketed"

[tool.sqlfluff.layout.type.column_constraint_segment]
spacing_before = "align"  # non-default spacing before column constraint
align_within = "create_table_statement"

[tool.sqlfluff.layout.type.comma]
line_position = "leading"

[tool.sqlfluff.layout.type.data_type]
spacing_before = "align"  # non-default spacing before data type
align_within = "create_table_statement"
```

Run the following command with sqlfluff:
```shell
$ sqlfluff lint --config pyproject.toml test.sql
```

### Dialect

databricks

### Version

```shell
$ sqlfluff --version
sqlfluff, version 3.4.2

$ python --version
Python 3.10.18
```

### Configuration

pyproject.toml:
```toml
[tool.sqlfluff.core]
# https://docs.sqlfluff.com/en/stable/reference/dialects.html
# ANSI is best for a general setting that is compatible with most SQL dialects.
# Modify this value if using a specific SQL dialect, e.g., databricks, postgres, sparksql, etc.
dialect = "databricks"  # For this project, we are using Databricks
large_file_skip_byte_limit = 0  # do not skip large files
max_line_length = 120
sql_file_exts = ".sql,.sql.j2,.dml,.ddl,.hql"

[tool.sqlfluff.indentation]
allow_implicit_indents = true
indented_on_contents = false

[tool.sqlfluff.layout.type.alias_expression]
spacing_before = "align"  # non-default spacing before alias expression
align_within = "select_clause"
align_scope = "bracketed"

[tool.sqlfluff.layout.type.column_constraint_segment]
spacing_before = "align"  # non-default spacing before column constraint
align_within = "create_table_statement"

[tool.sqlfluff.layout.type.comma]
line_position = "leading"

[tool.sqlfluff.layout.type.data_type]
spacing_before = "align"  # non-default spacing before data type
align_within = "create_table_statement"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
